### DISCUSSION TITLE
do not highlight DEBUG level log messages?

### DISCUSSION THREAD

**Discussion by jtmoon79** [Textualize/rich#3156] [Q&A]
I'm using `rich.logging.RichHandler` class to highlight a Python [`logging.Logger`](https://docs.python.org/3.8/library/logging.html#logger-objects), [e.g.](https://stackoverflow.com/a/68878216/471376)

```python
import logging
from rich.logging import RichHandler

FORMAT = "%(message)s"
logging.basicConfig(
    level=logging.DEBUG, format=FORMAT, datefmt="[%X]", handlers=[RichHandler()]
)
logger = logging.getLogger("rich")

logger.debug("debug...")
logger.info("info...")
logger.warning("warning...")
logger.error("error...")
logger.fatal("fatal...")
```

I want `DEBUG` log level messages to have no highlighting (or maybe even totally dimmed?).  This is so it's easier to visually skip past the less important messages.

Is it possible to not highlight (or dim) the `DEBUG` level log messages using the `RichHandler`?

**Comment by m1009d** [comment]
I am looking for this as well, I can't get this working. Can someone help please ? 

If I check the `python -m rich.theme`, I am afraid there has to be some hack because it's not possible to change the message per level:

```text
❯ python -m rich.theme
[styles]
...
log.level = none
log.message = none
log.path = dim
log.time = dim cyan
logging.keyword = bold yellow
logging.level.critical = bold reverse red
logging.level.debug = green
logging.level.error = bold red
logging.level.info = blue
logging.level.notset = dim
logging.level.warning = red
...
```

Changing the `log.message` will change all log lines.

**Comment by willmcgugan** [comment]
There is nothing builtin, but you could adapt or extend RichHandler. The code is quite short and straightforward.

**Reply by m1009d** [reply]
Unfortunately it's not so short for me , I am trying to understand the code but it's too complicated for me

**Comment by sr105** [comment]
I posted a solution to this in #3337

**Reply by m1009d** [reply]
Thanks a lot @sr105 , I will test this 👍🏻

**Reply by sr105** [reply]
I made the highlighter easier to reuse:

```python
    class StyledHighlighter(Highlighter):
        def __init__(self, *args, style=None, **kwargs) -> None:
            super().__init__(*args, **kwargs)
            assert type(style) is str, f"style must be a string! received {style!r}"
            self.style = style

        def highlight(self, text: Text) -> None:
            text.style = self.style

    def colored_error_filter(record: logging.LogRecord) -> (bool | logging.LogRecord):
        match record.levelno:
            case logging.ERROR:
                record.highlighter = StyledHighlighter(style="bold red")
                return record
            case logging.WARNING:
                record.highlighter = StyledHighlighter(style="bold orange1")
                return record
        return True
```
